File: financial-domain-starter/src/main/java/com/nixsolutions/financial/security/exception/InvalidTokenException.java
package com.nixsolutions.financial.security.exception;

public class InvalidTokenException extends RuntimeException
{
  public InvalidTokenException()
  {
    this("Token has invalid structure or is counterfeit");
  }

  public InvalidTokenException(String s)
  {
    super(s);
  }

  public InvalidTokenException(String s, Throwable throwable)
  {
    super(s, throwable);
  }

  public InvalidTokenException(Throwable throwable)
  {
    super(throwable);
  }
}

File: financial-domain-starter/src/main/kotlin/com/nixsolutions/financial/security/exception/InvalidTokenException.kt
package com.nixsolutions.financial.security.exception

import org.springframework.security.core.AuthenticationException

class InvalidTokenException : AuthenticationException {

  @JvmOverloads constructor(s: String = "Token has invalid structure or is counterfeit") : super(s)

  constructor(s: String, throwable: Throwable) : super(s, throwable)
}